src/precompile.jl

```diff
@@ -0,0 +1,47 @@
+using SnoopPrecompile: SnoopPrecompile
+
+const __TEST_NOTEBOOK_ID = uuid1()
+
+module __Foo end
+
+SnoopPrecompile.@precompile_all_calls begin
+    let
+        channel = Channel{Any}(10)
+        Pluto.PlutoRunner.setup_plutologger(
+            __TEST_NOTEBOOK_ID,
+            channel,
+        )
+    end
+    expr = Expr(:toplevel, :(1 + 1))
+    Pluto.PlutoRunner.run_expression(__Foo, expr, __TEST_NOTEBOOK_ID, uuid1(), nothing);
+
+    function wait_for_ready(notebook::Pluto.Notebook)
+        while notebook.process_status != Pluto.ProcessStatus.ready
+            sleep(0.1)
+        end
+    end
+
+    session = Pluto.ServerSession()
+    session.options.evaluation.workspace_use_distributed = false
+    basic = joinpath(pkgdir(Pluto), "sample", "Basic.jl")
+    nb = load_notebook(basic)
+
+    # Compiling PlutoRunner is very beneficial because it saves time in each notebook.
```

Comment:
But PlutoRunner is used by the notebook process, not the server process. Are we still getting a benefit?

Reply:
Maybe we need https://github.com/fonsp/Pluto.jl/pull/1881 ?

Reply:
> But PlutoRunner is used by the notebook process, not the server process. Are we still getting a benefit?

Ah! Because `Pluto` is not loaded in the notebook process, so the precompilation cache is also not loaded?

Yes maybe we need #1881 then indeed

Reply:
Yes it looks like precompiling `PlutoRunner` has no effect currently.

This is in the server process:
```julia
julia> using Pluto

julia> @time @eval Pluto.PlutoRunner.show_richest(IOBuffer(), (; n=1));
  0.009845 seconds (2.36 k allocations: 139.334 KiB, 85.56% compilation time)
```
and then after starting Pluto and running a new notebook with:
```
PlutoRunner.show_richest(IOBuffer(), (; n=1))
```
takes more than 1 seconds on the first run.